src/frontend/mame/infoxml.cpp

```diff
@@ -1809,6 +1815,40 @@ void output_driver(std::ostream &out, game_driver const &driver, device_t::featu
 	else
 		out << " savestate=\"unsupported\"";
 
+	if (flags & machine_flags::TYPE_ARCADE)
+		out << " type=\"arcade\"";
+	else if (flags & machine_flags::TYPE_CONSOLE)
+		out << " type=\"console\"";
+	else if (flags & machine_flags::TYPE_COMPUTER)
+		out << " type=\"computer\"";
+	else if (flags & machine_flags::TYPE_OTHER)
+		out << " type=\"other\"";
```

Comment:
Not exposing this was a conscious decision.  It’s poorly thought out, and a proper replacement for it presents a lot of practical issues.